### PR TITLE
Add multiple cdn

### PR DESCRIPTION
**Pull request by peng**
#### Description

Add `build.assetsPrefix` config docs if is object.

#### Related issues & labels (optional)

PR: https://github.com/withastro/astro/pull/10189

Issues discussed: https://github.com/withastro/astro/issues/9700#issuecomment-1907226632

Opened discussion: https://github.com/withastro/roadmap/discussions/818

### CODE REVIEW

File: src/content/docs/en/reference/configuration-reference.mdx
Comment (VoxelMC) [inline src/content/docs/en/reference/configuration-reference.mdx:504]:
```suggestion
<p>
**If this value is a `string`,**
</p>
```

If you word it like this, you will likely have to reword the following text below it to flow better.

File: src/content/docs/en/reference/configuration-reference.mdx
Comment (VoxelMC) [inline src/content/docs/en/reference/configuration-reference.mdx:521]:
```suggestion
<p>
**If this value is of `Record<string, string>`,**
</p>
```
Same for this

File: src/content/docs/en/reference/configuration-reference.mdx
Comment (VoxelMC) [inline src/content/docs/en/reference/configuration-reference.mdx:523]: Since this is documenting new behaviour, we will need to double check with core for accuracy!

File: src/content/docs/en/reference/configuration-reference.mdx
Comment (VoxelMC) [inline src/content/docs/en/reference/configuration-reference.mdx:495]:
```suggestion
**Type:** `string | Record.<string, string>`<br />
```

File: src/content/docs/es/reference/configuration-reference.mdx
Comment (VoxelMC) [inline src/content/docs/es/reference/configuration-reference.mdx:461]: See my note in my review comment about the translations!

**Review by VoxelMC — COMMENTED** [review]
Hello! Thank you for submitting these docs changes.
There are a few things to get out of the way first.
1. I see that this documents a currently proposed change! Once that change gets through scrutiny, this will be merged when it is appropriate :) (minor release day!)
2. In Astro Docs, we tend to write the `en` document *first*, merge it, *then* translate them. This way, it is less of a headache for the maintainers to check that every page is correct, and it's better for the translators to keep track of what is happening with the pages! You are more than welcome to PR the translations post-merge! I may be wrong about this, and if I am, let me know (maintainers). Just letting you know before moving forward!
3. We may need to go through some iterations to make sure that the documentation is just perfect! I, and I am sure others, would love to work with you to make that happen. 

Here's looking at getting that Astro PR merged, and this one, subsequently!

File: src/content/docs/en/reference/configuration-reference.mdx
Comment (VoxelMC) [inline src/content/docs/en/reference/configuration-reference.mdx:523]:
```suggestion
If your static files are served from different CDNs, you can configure Astro to use unique prefix paths for each CDN based on the filetype requested.
```

File: src/content/docs/en/reference/configuration-reference.mdx
Comment (VoxelMC) [inline src/content/docs/en/reference/configuration-reference.mdx:525]:
```suggestion
For example, if the static file is a `.js` file, it will be served from `https://js.cdn.example.com`. If it is a `.css` file, it will be served from `https://css.cdn.example.com`. All other assets will be served from `https://cdn.example.com`.
```
Some refactoring

File: src/content/docs/en/reference/configuration-reference.mdx
Comment (VoxelMC) [inline src/content/docs/en/reference/configuration-reference.mdx:532]:
```suggestion
      'css': 'https://css.cdn.example.com',
```
To match the example in the paragraph

File: src/content/docs/en/reference/configuration-reference.mdx
Comment (VoxelMC) [inline src/content/docs/en/reference/configuration-reference.mdx:539]:
```suggestion
The object key is the file extension, and its value is the CDN prefix path. You can set the default values of all other files by setting the value of `defaultAssetsPrefix`. If `defaultAssetsPrefix` is not set, when the file extension is not found in the `assetsPrefix` object, it will be set to the empty string `''` by default.
```
A little bit of rewording

**Review by VoxelMC — COMMENTED** [review]
This review comment is more for overall edits.
You have done a great job of writing out what this new change brings! Quite frankly, it's pretty cool :)
I have suggested some changes, but if you feel like they are incorrect, feel free to ignore them or discuss with me how we can meet somewhere in the middle!

Once more, thank you for working on this!!!

**Comment by sarah11918** [comment]
Hi @peng - Thank you for contributing!

This will take a little bit of work to resolve properly, because in fact this is not the appropriate place to make changes to the configuration reference file, nor should any changes be made to other languages. 

You should see at the top of this page when you edit it to tell you which file holds the source for this documentation:
image

Because this is also a feature / code change, the proper PR for this documentation is the same one you made your code changes in. That PR should also edit the above file and include documentation there, in proper JSDoc formatting. So, it's a little trickier because it must follow those syntax rules. You should look at other entries on the page and follow the model of how they are written.

Also, no changes should be made to any translated files. I'm assuming you do not speak all of these languages?  😄 We do not use any automated translations in our docs, and we have a system for our human translators, who are native speakers, to handle translations after a change is made to the English documentation. Most documentation contributions do require editing, so it is also inefficient to try to provide all languages up front before we have edited the English content.

So, this PR will eventually be closed. But, we can use this PR for some first edits, then you will know what to add to the other Astro PR and we can polish final documentation edits in that PR!

**Comment by sarah11918** [comment]
FYI - I removed the changes to the other languages so that I could remove our automated `i18n` label -- it will keep adding itself back as long as there is a change to a translated page, and then it's harder for me to find!  😄

File: src/content/docs/en/reference/configuration-reference.mdx
Comment (sarah11918) [reply src/content/docs/en/reference/configuration-reference.mdx:495]: I'm honestly not sure what's best to use here, but I don't see too many parentheses used, so I'll commit this and we'll figure out the tiny details when this is added to the other PR!  😄

File: src/content/docs/en/reference/configuration-reference.mdx
Comment (sarah11918) [inline src/content/docs/en/reference/configuration-reference.mdx:497]: Note to ourselves: this *property* was introduced in v2.2.0 but this *option* will be introduced much later. So we'll have to decide how to let people know, so that someone doesn't try to use this in an earlier version of Astro.

Comment (sarah11918) [reply src/content/docs/en/reference/configuration-reference.mdx:497]: Note also to ourselves: I think I have a solution I'm happy with. We can discuss to make sure!

**Review by sarah11918 — CHANGES_REQUESTED** [review]
Just putting a block on this to make sure it doesn't get merged here. This content should move to the other linked Astro PR.

**Comment by peng** [comment]
@sarah11918 OK, I will open another PR.

**Comment by peng** [comment]
@VoxelMC @sarah11918  I opened another PR: https://github.com/withastro/docs/pull/7183. Only en docs. Create by `astro/src/@types/astro.ts` file.

**Pull request closed without merging**
